### PR TITLE
if user haven't filled his informations yet. Click on a profile navigates to his own settings.

### PR DESCRIPTION
**Pull request by tisamo**
if user haven't filled his informations yet. Click on a profile navigates to his own settings.

### CODE REVIEW

File: client/src/app/shared/profilelist/profilelist.component.ts [hunk 1]
```diff
@@ -16,13 +18,23 @@ export class ProfilelistComponent implements OnInit {
...
               private profileListService: ProfilelistService,
-              private router: Router
-              ) {}
-  
-  
+              private router: Router,
+              private userDataService: UserDataService,
+              private userService: UserService) {
+    Object.assign(this.user, this.userService.getCurrentUser());
+    console.log(this.user);
```
Comment (ladape) [inline client/src/app/shared/profilelist/profilelist.component.ts]: please clear the code. remove console.logs

File: client/src/app/settings/settings.component.ts [hunk 2]
```diff
@@ -90,10 +93,11 @@ export class SettingsComponent implements OnInit {
     // this.settingsForm.valueChanges.subscribe(values => this.updateUser(values));
   }
 
-  logValue() {
-    console.log(this.interestForm.value);
+  noValues() {
+    if (this.ifUserhasValues) {
+      this.toastr.errorToastr('', 'Nem töltötte ki az összes adatot.');
```
Comment (ladape) [inline client/src/app/settings/settings.component.ts]: please use English

File: client/src/app/settings/settings.component.ts [hunk 3]
```diff
@@ -16,6 +17,7 @@ import {ToastrManager} from "ng6-toastr-notifications";
 })
 export class SettingsComponent implements OnInit {
   user: User = {} as User;
+  ifUserhasValues;
```
Comment (ladape) [inline client/src/app/settings/settings.component.ts]: if it is a boolean, please take care of name conversion, so modify it to isUserFilledDefaultValues.

File: client/src/app/settings/settings.component.ts [hunk 4]
```diff
@@ -90,8 +93,10 @@ export class SettingsComponent implements OnInit {
     // this.settingsForm.valueChanges.subscribe(values => this.updateUser(values));
   }
 
-  logValue() {
-    console.log(this.interestForm.value);
+  noValues() {
+    if (this.ifUserhasValues) {
+      this.toastr.errorToastr('', 'You haven' + 't' + ' ' + 'filled your credentials yet.');
```
Comment (ladape) [inline client/src/app/settings/settings.component.ts]: It is prettier   'You haven\'t filled y...

File: client/src/app/shared/profilelist/profilelist.component.html [hunk 5]
```diff
@@ -5,11 +5,11 @@
   <div class="container">
     <div class="row">
       <div class="col-md-4">
-        <mat-card *ngFor="let profile of profiles" [routerLink]="['/profile/'+ profile.userId]">
+        <mat-card *ngFor="let profile of profiles; let i = index" >
           <mat-card-header>
             <mat-card-title>{{profile.username}}</mat-card-title>
           </mat-card-header>
-          <img mat-card-image src="{{profile.avatar}}">
+          <img mat-card-image src="{{profile.avatar}}" (click)="getIndex(i)">
```
Comment (ladape) [inline client/src/app/shared/profilelist/profilelist.component.html]: It is wrong if for the first look it is not clean for me. getIndex? what is it doing? Please rename it with a more descriptive name.

File: client/src/app/shared/profilelist/profilelist.component.ts [hunk 6]
```diff
@@ -16,13 +18,35 @@ export class ProfilelistComponent implements OnInit {
...
-              private router: Router
-              ) {}
-  
-  
+              private router: Router,
+              private userDataService: UserDataService,
+              private userService: UserService) {
+    Object.assign(this.user, this.userService.getCurrentUser());
+    userDataService.ifUserHasValues = this.isEmpty();
+   console.log(this.isEmpty());
```
Comment (ladape) [inline client/src/app/shared/profilelist/profilelist.component.ts]: code formatting is wrong. Console log is not correct, remove it.

File: client/src/app/profile/profile.component.ts [hunk 7]
```diff
@@ -66,7 +67,7 @@ export class ProfileComponent implements OnInit {
     ).subscribe();
 
     this.currentUser.images.forEach(x => {
-      this.myImages.push(new myImage(x.url, x.url, 600, 600, 'zsa'));
+      this.myImages.push(new MyImage(x.url, x.url, 600, 600, 'zsa'));
```
[the remark this answers is not in the document]
Comment (tisamo) [reply client/src/app/profile/profile.component.ts]: It is the size of the modal which is created when you click on the image. I don't think its necessary save it to the database it functions more likely like a constant value

File: client/src/app/profile/avatar.service.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AvatarService {
+  avatarSrc = '';
```
[the remark this answers is not in the document]
Comment (tisamo) [reply client/src/app/profile/avatar.service.ts]: At the moment i think i need it

**Pull request merged**